projects/core/components/hosted-dropdown/hosted-dropdown.component.ts

```diff
@@ -43,9 +43,13 @@ export interface TuiHostedDropdownContext
     close(): void;
 }
 
-function shouldClose(this: TuiHostedDropdownComponent, {key}: KeyboardEvent): boolean {
+function shouldClose(
+    this: TuiHostedDropdownComponent,
+    event: Event | KeyboardEvent,
+): boolean {
     return (
-        key.toLowerCase() === 'escape' &&
+        'key' in event &&
+        event.key.toLowerCase() === 'escape' &&
```

Comment:
Maybe just `key?.toLowerCase()`? Or our new linter would say it's unnecessary? :D

Reply:
<img width="731" alt="image" src="https://user-images.githubusercontent.com/12021443/227190514-43c38e56-1ac4-4853-8125-f076d14b3b90.png">

Reply:
<img width="259" alt="image" src="https://user-images.githubusercontent.com/12021443/227190613-8a7b0d21-ba45-44a4-82cd-7d52fc9efba6.png">

although at the same time we will think that there can only be a line there and someone in the future will change it back from `key?.toLowerCase()` to `key.toLowerCase()`